document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -130,7 +130,26 @@ public CreateResult create(Key key, Document document) throws IOException {
 
   @Override
   public BulkUpdateResult bulkUpdate(List<BulkUpdateRequest> bulkUpdateRequests) throws Exception {
-    throw new UnsupportedOperationException();
+
+    var requestsWithFilter =
+        bulkUpdateRequests.stream()
+            .filter(req -> req.getFilter() != null)
+            .collect(Collectors.toList());
+
+    var requestsWithoutFilter =
+        bulkUpdateRequests.stream()
+            .filter(req -> req.getFilter() == null)
+            .collect(Collectors.toList());
+
+    int totalUpdateCountA = bulkUpdateRequestsWithFilter(requestsWithFilter);
```

Comment:
As discussed, let's go with the late exception approach.
- e.g if there are 10 documents to update, we will try for each document even if there is an expectation say updating `3rd` document.

Reply:
Done